starters/basic-starter/pages/[...slug].tsx

```diff
@@ -245,3 +249,48 @@ export async function getStaticProps(
     revalidate: 60,
   };
 }
+
+// Generates paths of pages to pre-render with prioritization of menu links.
+export async function generatePathsForBuild(context) {
+  let pathsFromContext = await drupalNoAuth.getStaticPathsFromContext(
+    ENTITY_TYPES,
+    context,
+  );
+  const menu = await drupalNoAuth.getMenu('main');
+
+  // Remove the '/' path from the menu items because it conflicts with [...slug].
+  const filteredMenuItems = menu.items.filter((item) => item.url !== '/');
+
+  // Generate paths from the menu links.
+  const pathsFromMenuItems = filteredMenuItems.map((item) => ({
+    params: { slug: item.url.split('/').slice(1) },
+  }));
+
+  const priorityMenuPaths = [];
+
+  // Remove the menu link paths from the pathsFromContext before moving them to the beginning.
+  pathsFromContext = pathsFromContext.filter((path) => {
+    if (shallowEqual(pathsFromMenuItems, path)) {
+      priorityMenuPaths.push(path);
```

Comment:
What do we need `priorityMenuPaths` for? Could we change this to a for loop and just remove the old array item here and add it to the beginning of the array?